perceval/backends/core/rss.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import json
+import logging
+import os.path
+
+import feedparser
+import requests
+
+from ...backend import Backend, BackendCommand, metadata
+from ...cache import Cache
+from ...errors import CacheError
+from ...utils import (DEFAULT_DATETIME,
+                      datetime_to_utc,
+                      str_to_datetime,
+                      urljoin)
+
+
+logger = logging.getLogger(__name__)
+
+
+class RSS(Backend):
+    """RSS backend for Perceval.
+
+    This class retrieves the entries from a RSS feed.
+    To initialize this class the URL must be provided.
+    The `url` will be set as the origin of the data.
+
+    :param url: RSS url
+    :param tag: label used to mark the data
+    :param cache: cache object to store raw data
+    """
+    version = '0.1.0'
+
+    def __init__(self, url, tag=None, cache=None):
+        origin = url
+
+        super().__init__(origin, tag=tag, cache=cache)
+        self.url = url
+        self.client = RSSClient(url)
+
+    @metadata
+    def fetch(self):
+        """Fetch the entries from the url.
+
+        The method retrieves all entries from a RSS url
+
+        :returns: a generator of entries
+        """
+
+        logger.info("Looking for rss entries at feed '%s'", self.url)
+
+        self._purge_cache_queue()
+        nentries = 0  # number of entries
+
+        raw_entries = self.client.get_entries()
+        self._push_cache_queue(raw_entries)
+        self._flush_cache_queue()
+        entries = feedparser.parse(raw_entries)['entries']
```

Comment:
Moved to a static method the parsing.

About the unit test, we are testing in this case the external library feedparser, right? Added!